### PR TITLE
Device pins

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/client/client.py [hunk 1]
```diff
@@ -391,3 +394,66 @@ def get_url(self, endpoint, *args):
...
+        :param limit:
+            The maximum number of entries to return per page. If not specified, then will use the server-side default.
+        :type limit:
+            `int` or None
+        :param marker:
+            The paging marker to start paging from.
+        :type marker:
+            `str` or None
+        :param direction:
+            The sorting direction.
```
Comment (jmoldow) [inline boxsdk/client/client.py]: Specify what the legal values are.

File: boxsdk/client/client.py [hunk 2]
```diff
@@ -391,3 +394,66 @@ def get_url(self, endpoint, *args):
...
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            An iterator of the entries in the device pins.
+        :rtype:
+            :class:`BoxObjectCollection`
+        """
+        additional_params = {
+            'direction': direction,
```
Comment (jmoldow) [inline boxsdk/client/client.py]: I assume that you need to not put `None` into here.

File: boxsdk/client/client.py [hunk 3]
```diff
@@ -391,3 +394,66 @@ def get_url(self, endpoint, *args):
...
+        :param device_pin_id:
+            The assignment ID of the :class:`DevicePin` object.
+        :type device_pin_id:
+            `unicode`
+        :return:
+            A :class `DevicePin` object with the given entry ID.
+        :rtype:
+            :class:`DevicePin`
+        """
+        return DevicePin(session=self._session, object_id=device_pin_id)
```
Comment (jmoldow) [inline boxsdk/client/client.py]: See the comments that I left on the legal holds PR.

File: boxsdk/object/device_pin.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from __future__ import unicode_literals
+
+from .base_object import BaseObject
+
+class DevicePin(BaseObject):
+    """Represents the device pinner"""
+
+
+    def get_url(self, *args):
+        return self._session.get_url('device_pinners', self._object_id, *args)
```
Comment (jmoldow) [inline boxsdk/object/device_pin.py]: Why did you need to add this? I don't think you should need it.

File: boxsdk/pagination/page.py [hunk 5]
```diff
@@ -38,7 +40,7 @@ def _translator(self):
 
         :rtype:   :class:`Translator`
         """
-        return self._session.translator
+        return Translator()
```
Comment (jmoldow) [inline boxsdk/pagination/page.py]: Revert this.
